File: Task1/binary.py
# -*- coding: utf-8 -*- 
# @Time : 2019/10/10 21:29 
# @Author : BaoBao
# @Mail : baobaotql@163.com 
# @File : binary.py 
# @Software: PyCharm

#recursion
def recursion(n):
    result = ''
    if n == 0:
        return result
    else:
        result = recursion(n//2)
        return result + str(n%2)

num_a = int(input("input a decimal number : "))

if num_a >= 90000:
    print("error！")
else:
    print(recursion(num_a))



#non-recursion

def non_recursion(n):
    s = ''
    while n > 0:
        s = str(n%2) + s    # 取余后更新 s
        n = n//2            # 取整后更新 n
    return s
num_b = int(input("input another decimal number : "))

if num_b >= 90000:
    print("error!")
else:
    print(non_recursion(num_b))

File: Task1/binary.cpp
#include<stdio.h>

int recursion(int n)
{
	int i;
	if(n==0)
	{
		return n;
	}
	else
	{
		recursion(n/2);
		printf("%d",n%2);	//因为输出在函数的后面 所以为逆序打印 
	}
 }  
 int non_resursion(int n)
 {
 	int s[100];
 	int i=0;
 	while(n>0)
 	{
 		s[i] = n%2;
 		i=i+1;
 		n=n/2;
	 }
	 for(i--;i>=0;i--)
	 {
	 	printf("%d", s[i]);
	 }
  } 
 
 int main()
 {
 	int num_1, num_2;
 	//递归 
 	printf("input a number : ");
 	scanf("%d", &num_1);
 	recursion(num_1);
 	printf("\r\n");
 	
 	//非递归 
 	printf("input a number : ");
 	scanf("%d", &num_2);
 	non_resursion(num_2);
 	
 	return 0;
 }
